compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/StableHLOCustomCalls.cpp

```diff
@@ -221,6 +310,242 @@ struct HouseholderReflectorRewriter final
+struct CholeskyRewriter final : OpRewritePattern<mlir::stablehlo::CholeskyOp> {
+  using OpRewritePattern<mlir::stablehlo::CholeskyOp>::OpRewritePattern;
+  using OpAdaptor = mlir::stablehlo::CustomCallOp::Adaptor;
+
+  LogicalResult matchAndRewrite(mlir::stablehlo::CholeskyOp op,
+                                PatternRewriter &rewriter) const final {
+    ImplicitLocOpBuilder b(op.getLoc(), rewriter);
+    auto operand = op.getOperand();
+    auto operandTy = cast<ShapedType>(operand.getType());
+    auto resultTy = cast<ShapedType>(op.getType());
+    const auto rank = operandTy.getRank();
+    if (!operandTy.hasStaticShape()) {
+      return rewriter.notifyMatchFailure(op,
+                                         "not supported for dynamic shapes");
+    }
+
+    Value left =
+        extractSlice(operand, makeIndexArray({0}, b), {1}, {rank - 1}, b);
+    Value topleft = extractSlice(operand, makeIndexArray({0, 0}, b), {1, 1},
+                                 {rank - 1, rank - 2}, b);
+    topleft = collapseBackDims(topleft, 2, b);
+
+    AffineMap topleftMap = getAffineMap({rank - 2, rank - 1}, rank, b);
+    AffineMap leftMap = b.getMultiDimIdentityMap(rank);
+    AffineMap outMap = b.getMultiDimIdentityMap(rank);
+
+    SmallVector<AffineMap> affineMaps = {topleftMap, leftMap, outMap};
+    SmallVector<utils::IteratorType> iterTypes(rank,
+                                               utils::IteratorType::parallel);
+    Value emptyCol = b.create<tensor::EmptyOp>(
+        cast<ShapedType>(left.getType()).getShape(), resultTy.getElementType());
+
+    Value generic = b.create<linalg::GenericOp>(
+                         emptyCol.getType(), ValueRange{topleft, left},
+                         emptyCol, affineMaps, iterTypes,
+                         [&](OpBuilder &bb, Location loc, ValueRange args) {
+                           Value sqrt = bb.create<math::SqrtOp>(loc, args[0]);
+                           Value div =
+                               bb.create<arith::DivFOp>(loc, args[1], sqrt);
+                           bb.create<linalg::YieldOp>(loc, div);
+                         })
+                        .getResult(0);
+
+    // We initialize the tensor with all zeros.
+    Value zeroF =
+        b.create<arith::ConstantOp>(b.getZeroAttr(resultTy.getElementType()));
+    Value empty = b.create<tensor::SplatOp>(resultTy, zeroF);
+
+    Value insert = insertSlice(empty, generic, {}, {}, b);
+
+    // Iterate row-by-row in the input matrix and upate the estimated
+    // cholesky decomposition.
+    Value one = b.create<arith::ConstantIndexOp>(1);
+    Value N = b.create<arith::ConstantIndexOp>(operandTy.getShape().back());
+    Value scf =
+        b.create<scf::ForOp>(
+             one, N, one, ValueRange{insert},
+             [&](OpBuilder &bb, Location loc, Value col, ValueRange args) {
+               ImplicitLocOpBuilder b(loc, bb);
+               // vecvec = dot(L[i, :], L[i, :]):
+               Value outSlice =
```

Comment:
L = args[0] ?

change comment to L[col, :]

At this spot, I'd not know why rank-2.